### PR TITLE
Add CodeQL template to arcade

### PR DESCRIPTION
**Pull request by dkurepa**
### To double check:

* [ ] The right tests are in and and the right validation has happened.  Guidance:  https://github.com/dotnet/arcade/tree/main/Documentation/Validation

https://github.com/dotnet/arcade/issues/13495
The PR adds a CodeQL template that other repos will be able to use by just supplying their build steps and the schedule.
CodeQL requires some variables to be set, and these need to be specified in the beginning of the top level yaml, hence the comments

### CODE REVIEW

**Comment by dkurepa** [comment]
This was tested on one of our new repos: https://dev.azure.com/dnceng/internal/_build/results?buildId=2180555&view=results

File: eng/common/templates/job/run-codeql.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+#   value: 0
+#   # CodeQL needs this plumbed along as a variable to enable TSA
+# - name: Codeql.TSAEnabled
+#   value: ${{ parameters.TSAEnabled }}
+#   # Variables for building
+# - name: _BuildConfig
+#   value: Release
+# - name: _InternalBuildArgs
+#   value: ''
+# - name: _ProductionArgs
```
Comment (andriipatsula) [inline eng/common/templates/job/run-codeql.yml:21]: 'd be nice to have a comment about `_InternalBuildArgs` and `_ProductionArgs` like you did for another parameters. Is it comma separated or semicolon separated values, etc. some example 'd be nice.

Comment (dkurepa) [reply eng/common/templates/job/run-codeql.yml:21]: Ah we actually don't want these here in the template, my bad. These are for a specific build you want to run. I'll remove them

Comment (dkurepa) [reply eng/common/templates/job/run-codeql.yml]: ok they're removed :D

**Comment by riarenas** [comment]
Is this going to be used by product repositories as well or only dnceng-owned repos? if it's the former, who is going to drive the usage of this template as opposed to our previous guidance?(https://github.com/dotnet/arcade/blob/main/Documentation/CodeQLGuidance.md)?

If it's the latter, eng/common is probably not the right place for this. 

@garath for thoughts as well.

**Comment by dkurepa** [comment]
> Is this going to be used by product repositories as well or only dnceng-owned repos? if it's the former, who is going to drive the usage of this template as opposed to our previous guidance which explicitly states Arcade is not providing a template for this (https://github.com/dotnet/arcade/blob/main/Documentation/CodeQLGuidance.md)?
> 
> If it's the latter, eng/common is probably not the right place for this.
> 
> @garath for thoughts as well.

It's the latter. We were creating codeql pipelines for the new repos and thought that this might be a good idea, while not knowing about the guidance. Is there some place else where we could do a thing like this? Or maybe we don't care too much, since it's not a lot of code to begin with

**Comment by premun** [comment]
@riarenas is the `eng/common` copying process hardcoded for Arcade in darc/Maestro or is that some configuration on the subscription?

**Comment by premun** [comment]
> It's an interesting thought if we need this kind of synchronization system for dnceng-owned code, especially since we're factoring out into more repos. Maybe what we should do is have one repo where we store these templates and we add that repo as a resource to the pipelines that need it? [learn.microsoft.com/en-us/azure/devops/pipelines/process/resources?view=azure-devops&tabs=schema](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/resources?view=azure-devops&tabs=schema)

I like the idea, maybe we'll find more uses for it. We'd need to define the right resource based on environment though (GH for public and AzDO for internal) which might be problematic a bit though. I think the YAML ifs are somehow restricted around `resources` as we've seen issues with this in the VMR templates. So we'd need a PoC for this part first.

**Comment by dkurepa** [comment]
I will close this PR. We can continue the discussion in the issue assigned to it

**Pull request closed without merging**